keps/sig-node/4639-oci-volume-source/README.md

```diff
@@ -0,0 +1,796 @@
+and share them among containers in a pod without including a shell in the image, thereby reducing vulnerabilities and simplifying image creation.
+
+## Motivation
+
+Currently, packaging files within images often requires the inclusion of a shell, even when the sole purpose of the image is to share files among containers in a pod.
+This not only increases the image size unnecessarily but also introduces potential security vulnerabilities. By supporting OCI images and artifacts
+directly as a `VolumeSource`, we can streamline the process and enhance security.
+
+### Goals
+
+- Introduce a new `VolumeSource` type that allows mounting OCI images and/or artifacts.
+- Simplify the process of sharing files among containers in a pod.
+- Reduce the need for including a shell in images, thereby minimizing security risks.
+- Reuse existing logic from ConfigMaps for managing file locations, garbage collection, and updates.
+
+### Non-Goals
+
+- This proposal does not aim to replace existing `VolumeSource` types.
+- This proposal does not address other use cases for OCI images beyond file sharing among containers in a pod.
+
+## Proposal
+
+We propose to add a new `VolumeSource` that supports OCI images and/or artifacts. This `VolumeSource` will allow users to mount an OCI image or artifact directly into a pod,
+making the files within the image accessible to the containers without the need for a shell or additional image manipulation.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a Kubernetes user, I want to share a configuration file among multiple containers in a pod without including a shell in my image, so that I can
+minimize security risks and image size. I want to package this file in an OCI object to take advantage of OCI distribution.
+
+#### Story 2
+
+As a DevOps engineer, I want to package and distribute binary artifacts using OCI images and mount them directly into my Kubernetes pods,
+so that I can streamline my CI/CD pipeline. I want to package this file in an OCI object to take advantage of OCI distribution.
+
+#### Story 3
+
+As a data scientist, MLOps engineer, or AI developer, I want to mount large language models or machine learning models in a pod alongside a model-server, so that I can efficiently serve the models
+without including them in the container image. I want to package these models in an OCI object to take advantage of OCI distribution and ensure efficient model deployment.
+
+### Notes/Constraints/Caveats (Optional)
+
+- This enhancement assumes that the cluster has access to the OCI registry.
+- The implementation must handle image pull secrets and other registry authentication mechanisms.
+- Performance considerations must be taken into account, especially for large images or artifacts.
+
+### Risks and Mitigations
+
+- **Security Risks:** Allowing direct mounting of OCI images introduces potential attack vectors. Mitigation includes thorough security reviews and limiting access to trusted registries.
```

Comment:
Also need to consider how this works with image pull secrets, especially with cached images that were pulled with secrets.